contracts/CBKLandSale.sol

```diff
@@ -0,0 +1,386 @@
+        availableLand[TIER_THREE]--;
+        chunkT3LandSoldTo[chunkId] = buyer;
+        chunkZoneLandSales[chunkIdToZoneId(chunkId)]++;
+
+        emit T3Given(buyer, chunkId);
+        cbkLand.mint(buyer, TIER_THREE, chunkId);
+    }
+
+    function giveT1LandFree(address buyer) public tierAvailable(TIER_ONE) restricted {
+        purchaseAddressMapping[buyer] = purchaseInfo(buyer, TIER_ONE, t1LandsSold, true);
+        sales[totalSales++] = purchaseAddressMapping[buyer];
+        emit T1GivenFree(buyer, t1LandsSold);
+        t1LandsSold++;
+        availableLand[TIER_ONE]--;
+        cbkLand.mint(buyer, TIER_ONE, 0);
+    }
+
+    function giveT2LandFree(address buyer, uint256 chunkId) public tierAvailable(TIER_TWO) chunkAvailable(chunkId) restricted {
+        // First t2 sale
+        if(chunkT2LandSales[chunkId] == 0){
+            chunksWithT2Land++;
+        }
+
+        t2LandsSold++;
+        chunkT2LandSales[chunkId]++;
+        chunkZoneLandSales[chunkIdToZoneId(chunkId)]++;
+
+        purchaseAddressMapping[buyer] = purchaseInfo(buyer, TIER_TWO, chunkId, true);
+        sales[totalSales++] = purchaseAddressMapping[buyer];
+        availableLand[TIER_TWO]--;
+
+        emit T2GivenFree(buyer, chunkId);
+        cbkLand.mint(buyer, TIER_TWO, chunkId);
+    }
+
+    function giveT3LandFree(address buyer, uint256 chunkId) public tierAvailable(TIER_THREE) chunkAvailable(chunkId) t3Available(chunkId) restricted {
+        t3LandsSold++;
+        
+        purchaseAddressMapping[buyer] = purchaseInfo(buyer, TIER_THREE, chunkId, true);
+        sales[totalSales++] = purchaseAddressMapping[buyer];
+        availableLand[TIER_THREE]--;
+        chunkT3LandSoldTo[chunkId] = buyer;
+        chunkZoneLandSales[chunkIdToZoneId(chunkId)]++;
+
+        emit T3GivenFree(buyer, chunkId);
+        cbkLand.mint(buyer, TIER_THREE, chunkId);
+    }
+
+    function chunkIdToZoneId(uint256 chunkId) internal pure returns (uint256){
+        return 10 * (chunkId / 1000) + (chunkId % 100) / 10;
+    }
+
+    function salesAllowed() public view returns (bool){
+        return _enabled;
+    }
+
+    function getAllowedLandOffset()  public view returns (uint256){
+        return _allowedLandOffset;
+    }
+
+    function checkIfChunkAvailable(uint256 tier, uint256 chunkId) public view returns (bool){
+        if(reservedChunkIds.contains(chunkId)){
+            return false;
+        }
+
+        if(chunkId > MAX_CHUNK_ID){
+            return false;
+        }
+
+        if(tier == TIER_ONE){
+            return availableLand[TIER_ONE] > 0;
+        }
+
+        if(tier == TIER_TWO){
+            return availableLand[TIER_TWO] > 0
+                    && chunkT2LandSales[TIER_TWO] < _allowedLandSalePerChunk
+                    && _chunkAvailableForT2(chunkId);
+        }
+
+        if(tier == TIER_THREE){
+            return availableLand[TIER_THREE] > 0
+                    && chunkT2LandSales[TIER_TWO] < _allowedLandSalePerChunk
```

Comment:
this stood out to me, is this a copy paste error, or is there a limit correlation between tier 2s and 3s?

Reply:
I guess you are correct. The only constraints on T3 purchase is enough total t3 land is available and the chunk itself has not had its t3 land pruchased.